keyboards/dztech/dz60rgb/rgbmatrix_layers.c

```diff
@@ -0,0 +1,40 @@
+#include QMK_KEYBOARD_H
+#include <rgblight_list.h>
+#include "dz60rgb.h"
+
+#define RGB_OFF 0x00, 0x00, 0x00
+
+// TODO: add proper defines so people can turn this all on and off
+// TODO: find a way to allow the user to customize colors (another enum?)
+
+
+// set weakly, can be overridden
+__attribute__ ((weak))
+void rgb_matrix_layer_helper (uint8_t red, uint8_t green, uint8_t blue, bool default_layer) {
+    for (int i = 0; i < DRIVER_LED_TOTAL; i++) {
+        if (HAS_FLAGS(g_led_config.flags[i], LED_FLAG_MODIFIER)) {
+            rgb_matrix_set_color( i, red, green, blue );
+        }
+    }
+}
+
+// set weakly, can be overridden
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void) {
+    uint8_t this_led = host_keyboard_leds();
+    if (!g_suspend_state) {
+        switch (biton32(layer_state)) {
+            case _LAYER1:
+            rgb_matrix_layer_helper(RGB_RED, false); break;
+            case _LAYER2:
+            rgb_matrix_layer_helper(RGB_GREEN, false); break;
+            case _LAYER3:
+            rgb_matrix_layer_helper(RGB_BLUE, false); break;
+            case _LAYER4:
+            rgb_matrix_layer_helper(RGB_YELLOW, false); break;
+        }
+    }
+    if ( this_led & (1<<USB_LED_CAPS_LOCK)) {
+        rgb_matrix_set_color(40, RGB_WHITE);
```

Comment:
Like above, I would suggest pulling index out into a const uin8_t or define. This allows for use in user keymaps a bit cleaner / easier.